executor/load_data.go

```diff
@@ -358,14 +392,48 @@ func (e *LoadDataWorker) Load(
 		importer.LogicalImportMode,
 		e.Ctx.GetSessionVars().User.String(),
 	)
+	if err != nil {
+		return 0, err
+	}
+
+	if e.controller.Detached {
+		go func() {
+			detachedCtx := context.Background()
+			detachedCtx = kv.WithInternalSourceType(detachedCtx, kv.InternalLoadData)
+			// error is stored in system table, so we can ignore it here
+			//nolint: errcheck
+			_ = e.doLoad(detachedCtx, readerInfos, jobID)
+		}()
+		return jobID, nil
+	}
+	return jobID, e.doLoad(ctx, readerInfos, jobID)
+}
+
+func (e *LoadDataWorker) doLoad(
+	ctx context.Context,
+	readerInfos []LoadDataReaderInfo,
+	jobID int64,
+) (err error) {
+	defer func() {
+		if e.ownSession {
+			e.putSysSessionFn(ctx, e.Ctx)
+		}
+	}()
+
+	s, err := e.getSysSessionFn()
 	if err != nil {
 		return err
 	}
+	defer e.putSysSessionFn(ctx, s)
+
+	sqlExec := s.(sqlexec.SQLExecutor)
 
 	defer func() {
+		ctx2 := context.Background()
```

Comment:
make it work even ctx is cancelled? maybe leave a comment

Reply:
https://github.com/pingcap/tidb/pull/42159/commits/4fa316edede8af4b70dd34949066f6967563a808